sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/SymmetricHashJoinStateManager.scala

```diff
@@ -767,6 +767,35 @@ object SymmetricHashJoinStateManager {
     }
   }
 
+  def getSchemaForStateStores(
+      joinSide: JoinSide,
+      inputValueAttributes: Seq[Attribute],
+      joinKeys: Seq[Expression],
+      stateFormatVersion: Int): scala.collection.mutable.Map[String, (StructType, StructType)] = {
```

Comment:
Is caller expected to modify the result? If not, let's simply return immutable Map.

Reply:
Done